File: server-scala-play-server/app/model/ListingResponse.scala
package model

import play.api.libs.json._

/**
  * Represents the Swagger definition for ListingResponse.
  */
@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
case class ListingResponse(
  id: Option[Long],
  name: Option[String],
  description: Option[String],
  start: Option[Long],
  end: Option[Long],
  locationName: Option[String],
  locationDescription: Option[String],
  privateListing: Option[Boolean],
  active: Option[Boolean],
  participants: Option[List[ParticipantResponse]],
  owner: Option[AccountShortResponse]
)

object ListingResponse {
  implicit lazy val listingResponseJsonFormat: Format[ListingResponse] = Json.format[ListingResponse]
}

File: server-kotlin-spring/src/main/kotlin/org/openapitools/model/ListingResponse.kt
package org.openapitools.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonProperty
import org.openapitools.model.AccountShortResponse
import org.openapitools.model.ParticipantResponse
import javax.validation.constraints.DecimalMax
import javax.validation.constraints.DecimalMin
import javax.validation.constraints.Email
import javax.validation.constraints.Max
import javax.validation.constraints.Min
import javax.validation.constraints.NotNull
import javax.validation.constraints.Pattern
import javax.validation.constraints.Size
import javax.validation.Valid
import io.swagger.v3.oas.annotations.media.Schema

/**
 * 
 * @param id 
 * @param name 
 * @param description 
 * @param start 
 * @param end 
 * @param locationName 
 * @param locationDescription 
 * @param privateListing 
 * @param active 
 * @param participants 
 * @param owner 
 */
data class ListingResponse(

    @Schema(example = "null", description = "")
    @get:JsonProperty("id") val id: kotlin.Long? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("name") val name: kotlin.String? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("description") val description: kotlin.String? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("start") val start: kotlin.Long? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("end") val end: kotlin.Long? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("locationName") val locationName: kotlin.String? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("locationDescription") val locationDescription: kotlin.String? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("privateListing") val privateListing: kotlin.Boolean? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("active") val active: kotlin.Boolean? = null,

    @field:Valid
    @Schema(example = "null", description = "")
    @get:JsonProperty("participants") val participants: kotlin.collections.List<ParticipantResponse>? = null,

    @field:Valid
    @Schema(example = "null", description = "")
    @get:JsonProperty("owner") val owner: AccountShortResponse? = null
) {

}

